scripts/publish.rs

```diff
@@ -0,0 +1,127 @@
+use std::env;
+use std::io::{self, Write};
+use std::process::{Command, Output};
+use std::str;
+
+// Crates.io API token
+const CRATES_IO_API_TOKEN: &str = "CRATES_IO_API_TOKEN";
+
+// Obtain local crate version
+fn local_version(crate_name: &str) -> String {
+    // Obtain local crate version contained in cargo pkgid data
+    let cargo_pkgid_output = Command::new("cargo")
+        .args(["pkgid", "-p", crate_name])
+        .output()
+        .expect("Failed to run cargo pkgid");
+
+    // Convert cargo pkgid output into a str
+    let cargo_pkgid_str = str::from_utf8(&cargo_pkgid_output.stdout)
+        .expect("Failed to convert pkgid output into a str");
+
+    // Extract only the local crate version from str
+    let (_, local_version) = cargo_pkgid_str
+        .split_once('#')
+        .expect("Failed to get local crate version");
+
+    local_version.trim_end().to_string()
+}
+
+// Obtain remote crate version
+fn remote_version(crate_name: &str) -> String {
+    // Obtain remote crate version contained in cargo search data
+    let cargo_search_output = Command::new("cargo")
+        .args(["search", crate_name, "--limit", "1"])
+        .output()
+        .expect("Failed to run cargo search");
+
+    // Convert cargo search output into a str
+    let remote_version_str = str::from_utf8(&cargo_search_output.stdout)
+        .expect("Failed to convert cargo search output into a str");
+
+    // Extract only the remote crate version from str
+    let (remote_version, _) = remote_version_str
+        .split_once('=')
+        .map(|(_, second)| second.trim_start().split_once(' '))
+        .flatten()
+        .expect("Failed to get remote crate version");
```

Comment:
The function shouldn't crash if there this is the first time being deployed, maybe returning `Option<String>` is preferable here and more explicit?

Reply:
Yes, you're right. I'm going to add a check

Reply:
I think it's better to check the results produced by `cargo search` in case of a non-existent crate and in that case return `None`